### PR TITLE
Refactor dataset retriever to isolate CLI functions

### PR DESCRIPTION
**Pull request by neubig**
# Description

In preparation for preparing our notebook demo (https://github.com/neulab/prompt2model/issues/289), it would be nice to split off the functions that require the CLI so we can replace them with more notebook-native ways of doing things. This PR starts by doing some refactoring to make the interface to the retriever a bit more cleanly separated.

# References

- Fixes: https://github.com/neulab/prompt2model/issues/292

# Blocked by

- NA

### CODE REVIEW

File: prompt2model/dataset_retriever/description_dataset_retriever.py [hunk 1]
```diff
@@ -3,51 +3,19 @@
...
-    """Read a string from stdin."""
-    description = str(input())
-    return description
-
-
-def input_y_n() -> bool:
-    """Get a yes/no answer from the user via stdin."""
-    y_n = str(input())
-    return not (y_n.strip() == "" or y_n.strip().lower() == "n")
+logger = logging.getLogger(__name__)
```
Comment (zhaochenyang20) [inline prompt2model/dataset_retriever/description_dataset_retriever.py:18]: I should change this in all the other components. 😅

File: prompt2model/dataset_retriever/description_dataset_retriever.py [hunk 2]
```diff
@@ -238,41 +232,34 @@ def canonicalize_dataset(self, dataset_name: str) -> datasets.DatasetDict:
...
+    ) -> list[DatasetInfo]:
+        """Retrieve the top datasets for a prompt.
 
         Args:
             prompt_spec: A prompt whose instruction field we use to retrieve datasets.
 
-        Return:
-            A list of relevant datasets dictionaries.
+        Returns:
+            A list of the top datasets for the prompt.
```
Comment (zhaochenyang20) [inline prompt2model/dataset_retriever/description_dataset_retriever.py:252]: Hmmm. The definition of top may be ambiguous. Should we summarize how we rank our datasets here?

**Review by zhaochenyang20 — APPROVED** [review]
A quick question:

Adding `_` to some function means that the function is private, right?

File: prompt2model/dataset_retriever/description_dataset_retriever.py (hunk 2 above)
Comment (neubig) [reply prompt2model/dataset_retriever/description_dataset_retriever.py]: Thanks! I added some more detail.

**Review by zhaochenyang20 — APPROVED** [review]
Go ahead!

**Pull request merged**